umi_tools/umi_methods.py

```diff
@@ -45,7 +45,9 @@ def get_umi_tag(read, tag='RX'):
     ''' extract the umi from the specified tag '''
 
     try:
-        return read.get_tag(tag).encode('utf-8')
+        # 10X pipelines append a 'GEM' tag to the UMI, e.g
+        # AGAGSGATAGATA-1
+        return read.get_tag(tag).split("-")[0].encode('utf-8')
```

Comment:
How much time does doing this to every read, irrespective of whether it contains a GEM or not add?